server_utils/origin.go

```diff
@@ -223,27 +231,103 @@ func GetOriginExports() (*[]OriginExport, error) {
+
+				if federationPrefix == "" {
+					return nil, errors.New(fmt.Sprintf("Volume mounts for S3 origins require a federation prefix, but none was provided: %s", volume))
+				}
+
+				if bucket == "" {
+					log.Warningf(`The volume mount %s does not contain a bucket. Pelican will interpret this as intending to export all buckets
+from S3 service URL. In this configuration, objects can be accessed at /federation/prefix/bucket/object`, volume)
+					log.Warningf(`This feature is only compatible with path-style URLs.`)
+				}
+
+				reads := param.Origin_EnableReads.GetBool() || param.Origin_EnablePublicReads.GetBool()
+				originExport := OriginExport{
+					FederationPrefix: federationPrefix,
+					StoragePrefix:    "/", // TODO: This is a placeholder for now, eventually we want storage prefix to mean something in S3
+					S3Bucket:         bucket,
+					S3AccessKeyfile:  param.Origin_S3AccessKeyfile.GetString(),
+					S3SecretKeyfile:  param.Origin_S3SecretKeyfile.GetString(),
+					Capabilities: server_structs.Capabilities{
+						PublicReads: param.Origin_EnablePublicReads.GetBool(),
+						Writes:      param.Origin_EnableWrites.GetBool(),
+						Listings:    param.Origin_EnableListings.GetBool(),
+						Reads:       reads,
+						DirectReads: param.Origin_EnableDirectReads.GetBool(),
+					},
+				}
+				*originExports = append(*originExports, originExport)
+			}
+
+			// If we're only exporting one namespace, we can set the internal Origin.FederationPrefix and Origin.StoragePrefix
+			if len(param.Origin_ExportVolumes.GetStringSlice()) == 1 {
+				viper.Set("Origin.FederationPrefix", (*originExports)[0].FederationPrefix)
+				viper.Set("Origin.S3Bucket", (*originExports)[0].S3Bucket)
+				viper.Set("Origin.EnableReads", (*originExports)[0].Capabilities.Reads)
+			}
+
+			log.Warningln("Passing export volumes via -v at the command line causes Pelican to ignore exports configured via the yaml file")
+			log.Warningln("However, namespaces exported this way will inherit the Origin.Enable* settings from your configuration")
+			log.Warningln("For finer-grained control of each export, please configure them in your pelican.yaml file")
+			return originExports, nil
 		}
 
-		viper.Set("Origin.FederationPrefix", federationPrefix)
-		*originExports = append(*originExports, originExport)
+		if param.Origin_Exports.IsSet() {
+			log.Infoln("Configuring multiple S3 exports from origin Exports block in config file")
+			if err := viper.UnmarshalKey("Origin.Exports", originExports, viper.DecodeHook(StringListToCapsHookFunc())); err != nil {
+				return nil, err
+			}
+			if len(*originExports) == 0 {
+				err := errors.New("Origin.Exports is defined, but no exports were found")
+				return nil, err
+			} else if len(*originExports) == 1 {
+				reads := (*originExports)[0].Capabilities.Reads || (*originExports)[0].Capabilities.PublicReads
+				viper.Set("Origin.FederationPrefix", (*originExports)[0].FederationPrefix)
```

Comment:
What happens if, say, one of these isn't set for originExports? Is there something that checks that the entries in originExports has every param needed?